lib/LDAPConnect.php

```diff
@@ -143,4 +146,38 @@ public function hasPasswordPolicy(): bool {
 		$ppDN = $this->ldapConfig->ldapDefaultPPolicyDN;
 		return !empty($ppDN);
 	}
+
+	/**
+	 * checks whether the LDAP server supports the passwd exop
+	 *
+	 * @param \LDAP\Connection $connection LDAP connection to check
+	 * @return boolean either the user can or cannot
+	 */
+	public function hasPasswdExopSupport($connection):bool {
+		if (is_null($this->passwdSupport)) {
```

Comment:
I’d really feel better if this is cached by connection, with current implementation if there are 2 ldap servers declared and only one supports the EXOP it will misbehave

Reply:
So make it a assoc. array with `spl_object_hash($connection)` as the key?

Reply:
No, would need to cache by prefix, but we do not have the prefix.
A refactor to properly manage multiple LDAP configurations in this app would be needed, let’s leave it aside for now.